pkg/apis/apps/v1beta2/defaults.go

```diff
@@ -29,17 +28,8 @@ func addDefaultingFuncs(scheme *runtime.Scheme) error {
 
 func SetDefaults_DaemonSet(obj *appsv1beta2.DaemonSet) {
 	labels := obj.Spec.Template.Labels
-
-	// TODO: support templates defined elsewhere when we support them in the API
-	if labels != nil {
-		if obj.Spec.Selector == nil {
-			obj.Spec.Selector = &metav1.LabelSelector{
-				MatchLabels: labels,
-			}
-		}
-		if len(obj.Labels) == 0 {
-			obj.Labels = labels
-		}
+	if labels != nil && len(obj.Labels) == 0 {
```

Comment:
Delete this. You are setting the labels of the DaemonSet to the labels of the PodTemplateSpec. You do not want to do this.